kit/Kit.cpp

```diff
@@ -1866,12 +1869,15 @@ class Document final : public DocumentManagerInterface
 
 static constexpr int traceEventRecordingsCapacity = 100;
 static std::vector<std::string> traceEventRecordings;
+static std::mutex traceEventRecordingsMutex;
 
 static void flushTraceEventRecordings()
 {
     if (traceEventRecordings.size() == 0)
         return;
 
+    assert(mainKitSocketPollThread == std::this_thread::get_id());
```

Comment:
Surely this also needs to take the mutex ?

Reply:
Yes.